doc/api/path.md

```diff
@@ -177,57 +247,73 @@ path.isAbsolute('bar\\baz')  // false
 path.isAbsolute('.')         // false
 ```
 
-*Note:* If the path string passed as parameter is a zero-length string, unlike
-        other path module functions, it will be used as-is and `false` will be
-        returned.
+A [`TypeError`][] is thrown if `path` is not a string.
 
 ## path.join([path1][, path2][, ...])
 
-Join all arguments together and normalize the resulting path.
+* `path1` {String} A path segment
+* `path2` {String} A path segment
+* `[, ...]` {String} Additional path segments
+
+The `path.join()` method join all given path segments together using the
+platform specific separator (`path.sep`) as a delimiter, then normalizes
```

Comment:
Referencing `path.sep` specifically here may be a bit misleading in case someone tries to modify `path.sep`, since `path.sep` is not actually used internally by the path functions.